Fix: Bowtie2 installs into /usr/local/bin

hartzell opened the pull request:
The bowtie2 Makefile uses `prefix`, not `PREFIX`.  As is the package was trying to install into `/usr/local/bin`.  On my AWS instance I'm not allowed to do that, in a Docker container that succeeded but the Spack sanity check failed with:

```
==> Error: InstallError: Install failed for bowtie2.  Nothing was installed!
```

Tested on CentOS 7.

hartzell committed:
Update package.py

The bowtie2 Makefile uses `prefix`, not `PREFIX`.  As is the package was trying to install into `/usr/local/bin`.  On my AWS instance I'm not allowed to do that, in a Docker container that succeeded but the Spack sanity check failed with:

```
==> Error: InstallError: Install failed for bowtie2.  Nothing was installed!
```

tkameyama commented on var/spack/repos/builtin/packages/bowtie2/package.py:
```diff
@@ -66,4 +66,4 @@ def build_targets(self):
     @property
     def install_targets(self):
-        return ['PREFIX={0}'.format(self.prefix), 'install']
+        return ['prefix={0}'.format(self.prefix), 'install']
```
```suggestion
        if self.spec.satisfies('@:2.3.9'):
            return ['prefix={0}'.format(self.prefix), 'install']
        else:
            return ['PREFIX={0}'.format(self.prefix), 'install']
```
the Makefile variable is changed from prefix to PREFIX in 2.4.0 beta2 .
https://github.com/BenLangmead/bowtie2/commit/8d123c66c1f637604046add1a15120ad04060ec0

hartzell replied:
@tkameyama -- your change also works for me w/ `@2.3.5.1`.

:satisfied:

hartzell replied:
@tkameyama -- would you like to fix this or would you like me to update this PR to use your patch?

tkameyama replied:
Please update to you.

hartzell replied:
done.

Commit:
Handle case change of install target "prefix"

Deal with the fact that in releases before 2.4 the Makefile used
'prefix', after it used 'PREFIX'.

Credit to @tkameyama
